modmesh/system.py

```diff
@@ -0,0 +1,130 @@
+#   may be used to endorse or promote products derived from this software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+
+"""
+Runtime system code
+"""
+
+
+import builtins
+import sys
+import os
+import argparse
+
+import modmesh
+from . import view
+
+
+__all__ = [
+    'setup_process',
+]
+
+
+class ModmeshArgumentParser(argparse.ArgumentParser):
+    def __init__(self, *args, **kw):
+        super().__init__(*args, **kw)
+        self.exited = False
+        self.exited_status = None
+        self.exited_message = None
+
+    def exit(self, status=0, message=None):
+        self.exited = True
+        self.exited_status = status
+        self.exited_message = message
+        return
+
+
+def _parse_command_line(argv):
+    parser = ModmeshArgumentParser(description="Viewer")
+    parser.add_argument('--mode', dest='mode', action='store',
+                        default='viewer',
+                        choices=['viewer', 'command', 'terminal', 'pytest'],
+                        help='mode selection (default = %(default)s)')
+    parser.add_argument('--command', dest='command', action='store',
+                        default='', help='command to run')
+    args = parser.parse_args(argv[1:])
+    if parser.exited:
+        args.exit = (parser.exited_status, parser.exited_message)
+    else:
+        args.exit = tuple()
+    return args
+
+
+def _setup_builtin():
+    """Install the namespace."""
+    builtins.modmesh = modmesh
+    builtins.mm = modmesh
+
+
+def _run_viewer(argv):
+    """Run the viewer application."""
+    view.app().setup()
+    return view.app().exec()
+
+
+def _run_ipython(argv):
+    # Import IPython locally to avoid making it a dependency to the whole
+    # modmesh.
+    import IPython
+    ret = IPython.start_ipython(argv=[])
+    return ret
+
+
+def _run_command(cmd):
+    exec(cmd)
+    return 0
+
+
+def _run_pytest():
+    # Import pytest locally to avoid making it a dependency to the whole
+    # modmesh.
+    import pytest
+    mmpath = os.path.join(os.path.dirname(modmesh.__file__), '..', 'tests')
+    mmpath = os.path.abspath(mmpath)
+    return pytest.main(['-v', '-x', mmpath])
+
+
+def setup_process(argv):
```

Comment:
`setup_process` is probably not a good name because the role is the main function.  But let's use it for now.